cmds/jfrogplugin/uploaders/helm/upload.go

```diff
@@ -0,0 +1,105 @@
+package helm
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"path"
+	"strings"
+
+	"ocm.software/ocm/api/credentials"
+	"ocm.software/ocm/api/ocm/extensions/accessmethods/helm"
+	"ocm.software/ocm/api/ocm/plugin/ppi"
+)
+
+func Upload(
+	ctx context.Context,
+	data io.Reader,
+	client *http.Client,
+	url *url.URL,
+	creds credentials.Credentials,
+) (_ ppi.AccessSpec, err error) {
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	var req *http.Request
+	var res *http.Response
+	if req, err = http.NewRequestWithContext(ctx, http.MethodPut, url.String(), data); err != nil {
+		return nil, fmt.Errorf("failed to create HTTP request for upload: %w", err)
+	}
+	SetHeadersFromCredentials(req, creds)
+
+	if res, err = client.Do(req); err != nil {
+		return nil, fmt.Errorf("failed to store blob in artifactory: %w", err)
+	}
+	defer func() {
+		err = errors.Join(err, res.Body.Close())
+	}()
+
+	if invalid := 200 > res.StatusCode || res.StatusCode >= 300; invalid {
```

Comment:
302 is redirect that can sometimes happen with proxies. Are we concerned about that at all?

Reply:
the default http transport should already follow redirects. I believe this should cover this scenario: https://golang.org/pkg/net/http/#Client